MediaBrowser.Providers/TV/SeriesMetadataService.cs

```diff
@@ -88,18 +88,22 @@ protected override void MergeData(MetadataResult<Series> source, MetadataResult<
 
             var sourceItem = source.Item;
             var targetItem = target.Item;
-            var sourceSeasonNames = sourceItem.SeasonNames;
-            var targetSeasonNames = targetItem.SeasonNames;
+            var sourceSeasonNames = sourceItem.GetSeasonNames();
+            var targetSeasonNames = targetItem.GetSeasonNames();
 
-            if (replaceData || targetSeasonNames.Count == 0)
+            if (replaceData)
             {
-                targetItem.SeasonNames = sourceSeasonNames;
+                foreach (var (number, name) in sourceSeasonNames)
+                {
+                    targetItem.SetSeasonName(number, name);
+                }
             }
-            else if (targetSeasonNames.Count != sourceSeasonNames.Count || !sourceSeasonNames.Keys.All(targetSeasonNames.ContainsKey))
+            else if (!sourceSeasonNames.Keys.All(targetSeasonNames.ContainsKey))
```

Comment:
You're iterating twice in worst case. I would just cut out the first iteration
```suggestion
            else
```